src/components/RequestDashboard/TasksSection.jsx

```diff
@@ -0,0 +1,277 @@
+  };
+
+  const tryDelete = task => {
+    setTaskToDelete(task);
+    setOpen(true);
+  };
+  const assignTask = task => {
+    // TODO: should allow assigning to anybody, not just self
+    if (task) {
+      let user = props.client.user.id;
+      if (!user) {
+        user = `Practitioner/${state.defaultUser}`;
+      }
+      task.owner = {
+        reference: user
+      };
+      if (task.for) {
+        // reset 'for' element before updating
+        task.for = {
+          reference: `${task.for.resourceType}/${task.for.id}`
+        };
+      }
+      props.client.update(task).then(e => {
+        fetchTasks();
+      });
+    }
+  };
+  const deleteTask = () => {
+    if (taskToDelete) {
+      props.client.delete(`${taskToDelete.resourceType}/${taskToDelete.id}`).then(e => {
+        console.log('Deleted Task');
+        fetchTasks();
+      });
+      setOpen(false);
+      setTaskToDelete('');
+    }
+  };
+  const fetchTasks = () => {
+    let identifier = 'Task';
+    if (state.patient && state.patient.id) {
+      identifier = `Task?patient=${state.patient.id}`;
+    }
+    props.client.request(identifier, { resolveReferences: ['for', 'owner'] }).then(request => {
+      console.log(request);
+      if (request && request.entry) {
+        setTasks(request.entry.map(e => e.resource));
+      } else {
+        setTasks([]);
+      }
+    });
+  };
+  useEffect(() => {
+    fetchTasks();
+  }, []);
+
+  useEffect(() => {
+    fetchTasks();
+  }, [state.patient]);
+  const renderTasks = taskSubset => {
+    if (taskSubset.length > 0) {
+      return (
+        <Grid spacing={2} container>
+          {taskSubset.map(t => renderTask(t))}
+        </Grid>
+      );
+    } else {
+      return <p className={classes.noTasks}>No Tasks Found</p>;
+    }
+  };
+  const renderTask = task => {
+    let statusColor = '#555';
+    if (task.status.toLowerCase() === 'ready') {
+      statusColor = '#198754';
+    }
+
+    let taskForName = 'N/A';
+    let taskOwnerName = 'N/A';
+    if (task.for.resourceType.toLowerCase() === 'patient') {
+      const patient = task.for;
+      if (patient.name) {
+        taskForName = `${patient.name[0].given[0]} ${patient.name[0].family}`;
+      }
+    }
+    if (task.owner && task.owner.resourceType.toLowerCase() === 'practitioner') {
+      const practitioner = task.owner;
+      if (practitioner.name) {
+        taskOwnerName = `${practitioner.name[0].given[0]} ${practitioner.name[0].family}`;
+      } else {
+        taskOwnerName = task.owner.id;
+      }
+    }
+    let ownerText = (
+      <Stack direction="row" alignItems="center" gap={1}>
+        <AssignmentLateIcon />
+        Unassigned
+      </Stack>
+    );
+    if (task.owner) {
+      ownerText = (
+        <Stack sx={{ color: '#333' }} direction="row" alignItems="center" gap={1}>
+          <AssignmentTurnedInIcon />
+          {`Assigned to ${taskOwnerName}`}
+        </Stack>
+      );
+    }
+    return (
+      <Fragment key={task.id}>
```

Comment:
I think the contrast between the task's UI element background color (very light gray) and the white background of the page is too low. Could you style the task grid (or use a Card) so it has a drop-shadow around the top/left/right edges like a MUI 5 Card?

Also, the font-size for the Task ID, Timestamp, Beneficiary, and STATUS is rather tiny on large/medium screen sizes (I'm using a 1920x1080 monitor).

https://mui.com/material-ui/react-card/

![image](https://github.com/mcode/request-generator/assets/33106214/992e84b5-cfff-4e27-8615-e6b260aeb50b)

![image](https://github.com/mcode/request-generator/assets/33106214/185f8d3e-342d-450e-83fe-411e3d87cc9c)

Reply:
the header for the task is all unimportant information, it's just there in case it's needed, but is supposed to be small as to avoid taking any focus.  The status is arguably important but it will be moved elsewhere in another ticket